client/app/controllers/collection/add.js

```diff
@@ -150,15 +188,43 @@ export default Ember.Controller.extend({
 
     // Delete widget and resets state
     delete_widget_signature: ['widget_object'],
-    delete_widget: function(widget_object) {
+    delete_widget: function(widget_object, parameters) {
         this.get('widgets').removeObject(widget_object.value);
-        widget_object.value = undefined;
-        widget_object.state = ['undefined'];
+        Ember.set(widget_object, 'value', undefined);
+        Ember.set(widget_object, 'state', ['undefined']);
+    },
+
+
+    disableWidget_signature: ['widget_object'],
```

Comment:
Let's stick to camelCase for all JS variable & function names, so in this case `disableWidgetSignature` instead of `disableWidget_signature`